solr/solrj/src/java/org/apache/solr/client/solrj/impl/ConcurrentUpdateSolrClient.java

```diff
@@ -116,12 +114,23 @@ protected ConcurrentUpdateSolrClient(Builder builder) {
     this.streamDeletes = builder.streamDeletes;
     this.connectionTimeout = Math.toIntExact(builder.connectionTimeoutMillis);
     this.soTimeout = Math.toIntExact(builder.socketTimeoutMillis);
-    this.stallTime = Integer.getInteger("solr.cloud.client.stallTime", 15000);
-    if (stallTime < pollQueueTime * 2) {
+    this.pollQueueTimeMillis = builder.pollQueueTime;
+    this.stallTimeMillis = Integer.getInteger("solr.cloud.client.stallTime", 15000);
+
+    // make sure the stall time is larger than the polling time
+    // to give a chance for the queue to change
+    int minimalStallTime = pollQueueTimeMillis * 2;
+    if (minimalStallTime > this.stallTimeMillis) {
+      this.stallTimeMillis = minimalStallTime;
+    }
```

Comment:
Should these changes, and the one below, be in a separate PR?

Reply:
What I found was that the deprecated method setPollQueueTime is being used..  and that the logic iin the method needed to be preserved, so then I moved it...   I did want to have an extra set of eyes on it ;-).    WOuld you like me to cherry-pick the commit into a new PR???

Reply:
Ok, I saw that now. Looks good.

Reply:
Awesome!